pkg/scheduler/scheduler_test.go

```diff
@@ -1110,6 +1209,9 @@ func TestSchedule(t *testing.T) {
 			defer cancel()
 
 			scheduler.schedule(ctx)
+			if tc.needScheduleTwice {
```

Comment:
Instead of doing this, why not put the test case in `TestLastSchedulingContext`, which already does 2 attempts.